packit/utils/obs_helper.py

```diff
@@ -0,0 +1,287 @@
+    root = ET.Element("project")
+    root.attrib["name"] = project_name
+
+    (title_elem := ET.Element("title")).text = "Packit project"
+    (descr_elem := ET.Element("description")).text = description or ""
+    (person_elem := ET.Element("person")).attrib["userid"] = owner
+    person_elem.attrib["role"] = "maintainer"
+
+    for elem in (title_elem, descr_elem, person_elem):
+        root.append(elem)
+
+    for repo in repos:
+        (repo_elem := ET.Element("repository")).attrib["name"] = repo.name
+
+        for path_entry in repo.path:
+            (path_elem := ET.Element("path")).attrib["project"] = path_entry.project
+            path_elem.attrib["repository"] = path_entry.repository
+            repo_elem.append(path_elem)
+
+        for arch in repo.arch:
+            (arch_elem := ET.Element("arch")).text = arch
+            repo_elem.append(arch_elem)
+
+        root.append(repo_elem)
+
+    return root
+
+
+def create_package(project_name: str, package_name: str) -> None:
+    """Creates the package with the name ``package_name`` in the project
+    ``project_name`` on OBS. No sources are uploaded in the process.
+
+    """
+    (root := ET.Element("package")).attrib["name"] = package_name
+    root.attrib["project"] = project_name
+
+    (title := ET.Element("title")).text = f"The {package_name} package"
+    descr = ET.Element("description")
+
+    root.append(title)
+    root.append(descr)
+
+    package_url = core.makeurl(
+        _API_URL,
+        ["source", project_name, package_name, "_meta"],
+    )
+    metafile = core.metafile(package_url, ET.tostring(root))
+    metafile.sync()
+
+
+def create_obs_project(
+    project: str,
+    targets: str,
+    owner: Optional[str],
+    package_config: PackageConfig,
+    description: Optional[str],
+):
+    conf.get_config()
+    owner = owner or conf.config["api_host_options"][_API_URL]["user"]
+    project_name = project or f"home:{owner}:packit"
+
+    targets_list = targets.split(",")
+    for target in targets_list:
+        if target in DEPRECATED_TARGET_MAP:
+            logger.warning(
+                f"Target '{target}' is deprecated. "
+                f"Please use '{DEPRECATED_TARGET_MAP[target]}' instead.",
+            )
+
+    project_metadata = targets_to_project_meta(
+        targets=targets_list,
+        owner=owner,
+        project_name=project_name,
+        description=description,
+    )
+
+    logger.info(f"Using OBS project name = {project_name}")
+
+    project_url = core.makeurl(
+        _API_URL,
+        ["source", project_name, "_meta"],
+    )
+    metafile = core.metafile(project_url, ET.tostring(project_metadata))
+    metafile.sync()
+
+    package_names = list(package_config.packages.keys())
+
+    if len(package_names) != 1:
+        raise ValueError("Cannot handle multiple packages in package_config")
+
+    create_package(project_name, (package_name := package_names[0]))
+
+    return project_name, package_name
+
+
+def init_project(
```

Comment:
Sorry about being picky about the names..;).. but can we use `init_obs_project` for consistency with `create_obs_project`?